homeassistant/components/enocean/binary_sensor.py

```diff
@@ -97,8 +104,10 @@ def value_changed(self, packet):
         elif action == 0x15:
             self.which = 10
             self.onoff = 1
+
         self.hass.bus.fire(EVENT_BUTTON_PRESSED,
                            {'id': self.dev_id,
-                            'pushed': pushed,
+                            'pushed': self.pushed,
```

Comment:
We shouldn't change the event data until the event firing has been moved from the binary sensor platform to the enocean component. Platforms shouldn't fire events besides the state changed events that they fire by default when implemented correctly.

Reply:
I disagree with that statement (at this code location), if the value is always wrong. This is clearly a bugfix. 
I agree on setting `_state` though. I think it's even wrong, as most enocean rocker switches have two switches, which would make the value unreliable anyway. This can only be fixed by moving it to the enocean component (which I really should start doing soon, huh!)